File: base/--build/tmp/kapt3/stubs/debug/com/shopify/shopifyapp/basesection/viewmodels/LeftMenuViewModel.java
package com.baytrendy.shopifyapp.basesection.viewmodels;

import java.lang.System;

@kotlin.Metadata(mv = {1, 1, 16}, bv = {1, 0, 3}, k = 1, d1 = {"\u0000r\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\b\n\u0002\b\u0003\n\u0002\u0018\u0002\n\u0002\u0010 \n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\u0010\u000e\n\u0002\b\u0003\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u000b\n\u0002\b\u0004\n\u0002\u0018\u0002\n\u0002\b\u0004\n\u0002\u0010\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0005\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0005\u0018\u00002\u00020\u0001B\r\u0012\u0006\u0010\u0002\u001a\u00020\u0003\u00a2\u0006\u0002\u0010\u0004J\f\u0010\u001f\u001a\b\u0012\u0004\u0012\u00020\u001c0\nJ\u0010\u0010 \u001a\u00020!2\u0006\u0010\"\u001a\u00020#H\u0002J\u0012\u0010$\u001a\u000e\u0012\n\u0012\b\u0012\u0004\u0012\u00020\f0\u000b0\nJ\u0006\u0010%\u001a\u00020!J\b\u0010&\u001a\u00020!H\u0002J\b\u0010\'\u001a\u00020!H\u0002J\u000e\u0010(\u001a\u00020!2\u0006\u0010\r\u001a\u00020)J\u0017\u0010*\u001a\u00020!2\f\u0010+\u001a\b\u0012\u0004\u0012\u00020-0,H\u0082\u0002J\u0006\u0010.\u001a\u00020!J\b\u0010/\u001a\u00020!H\u0014J\u0010\u00100\u001a\u00020!2\b\u00101\u001a\u0004\u0018\u00010\u000fR\u0011\u0010\u0005\u001a\u00020\u00068F\u00a2\u0006\u0006\u001a\u0004\b\u0007\u0010\bR\u001a\u0010\t\u001a\u000e\u0012\n\u0012\b\u0012\u0004\u0012\u00020\f0\u000b0\nX\u0082\u0004\u00a2\u0006\u0002\n\u0000R#\u0010\r\u001a\u0014\u0012\u0010\u0012\u000e\u0012\u0004\u0012\u00020\u000f\u0012\u0004\u0012\u00020\u000f0\u000e0\n\u00a2\u0006\b\n\u0000\u001a\u0004\b\u0010\u0010\u0011R\u000e\u0010\u0012\u001a\u00020\u0013X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u0014\u001a\u00020\u0015X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u0011\u0010\u0016\u001a\u00020\u00178F\u00a2\u0006\u0006\u001a\u0004\b\u0016\u0010\u0018R\u0017\u0010\u0019\u001a\b\u0012\u0004\u0012\u00020\u000f0\n\u00a2\u0006\b\n\u0000\u001a\u0004\b\u001a\u0010\u0011R\u000e\u0010\u0002\u001a\u00020\u0003X\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u0014\u0010\u001b\u001a\b\u0012\u0004\u0012\u00020\u001c0\nX\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u0011\u0010\u001d\u001a\u00020\u00068F\u00a2\u0006\u0006\u001a\u0004\b\u001e\u0010\b\u00a8\u00062"}, d2 = {"Lcom/shopify/shopifyapp/basesection/viewmodels/LeftMenuViewModel;", "Landroidx/lifecycle/ViewModel;", "repository", "Lcom/shopify/shopifyapp/repositories/Repository;", "(Lcom/shopify/shopifyapp/repositories/Repository;)V", "cartCount", "", "getCartCount", "()I", "currencyResponseLiveData", "Landroidx/lifecycle/MutableLiveData;", "", "Lcom/shopify/buy3/Storefront$CurrencyCode;", "data", "Ljava/util/HashMap;", "", "getData", "()Landroidx/lifecycle/MutableLiveData;", "disposables", "Lio/reactivex/disposables/CompositeDisposable;", "handler", "Landroid/os/Handler;", "isLoggedIn", "", "()Z", "message", "getMessage", "responseLiveData", "Lcom/shopify/shopifyapp/utils/ApiResponse;", "wishListcount", "getWishListcount", "Response", "consumeResponse", "", "reponse", "Lcom/shopify/shopifyapp/utils/GraphQLResponse;", "currencyResponse", "fetchUserData", "getCurrency", "getMenus", "insertPreviewData", "Lorg/json/JSONObject;", "invoke", "result", "Lcom/shopify/buy3/GraphCallResult;", "Lcom/shopify/buy3/Storefront$QueryRoot;", "logOut", "onCleared", "setCurrencyData", "currencyCode", "base_debug"})
public final class LeftMenuViewModel extends androidx.lifecycle.ViewModel {
    private final io.reactivex.disposables.CompositeDisposable disposables = null;
    private final androidx.lifecycle.MutableLiveData<com.baytrendy.shopifyapp.utils.ApiResponse> responseLiveData = null;
    @org.jetbrains.annotations.NotNull()
    private final androidx.lifecycle.MutableLiveData<java.lang.String> message = null;
    @org.jetbrains.annotations.NotNull()
    private final androidx.lifecycle.MutableLiveData<java.util.HashMap<java.lang.String, java.lang.String>> data = null;
    private final androidx.lifecycle.MutableLiveData<java.util.List<com.shopify.buy3.Storefront.CurrencyCode>> currencyResponseLiveData = null;
    private final android.os.Handler handler = null;
    private final com.baytrendy.shopifyapp.repositories.Repository repository = null;
    
    @org.jetbrains.annotations.NotNull()
    public final androidx.lifecycle.MutableLiveData<java.lang.String> getMessage() {
        return null;
    }
    
    @org.jetbrains.annotations.NotNull()
    public final androidx.lifecycle.MutableLiveData<java.util.HashMap<java.lang.String, java.lang.String>> getData() {
        return null;
    }
    
    public final boolean isLoggedIn() {
        return false;
    }
    
    @org.jetbrains.annotations.NotNull()
    public final androidx.lifecycle.MutableLiveData<com.baytrendy.shopifyapp.utils.ApiResponse> Response() {
        return null;
    }
    
    public final int getCartCount() {
        return 0;
    }
    
    public final int getWishListcount() {
        return 0;
    }
    
    public final void fetchUserData() {
    }
    
    private final void getMenus() {
    }
    
    @org.jetbrains.annotations.NotNull()
    public final androidx.lifecycle.MutableLiveData<java.util.List<com.shopify.buy3.Storefront.CurrencyCode>> currencyResponse() {
        return null;
    }
    
    private final void getCurrency() {
    }
    
    @java.lang.Override()
    protected void onCleared() {
    }
    
    private final void invoke(com.shopify.buy3.GraphCallResult<? extends com.shopify.buy3.Storefront.QueryRoot> result) {
    }
    
    private final void consumeResponse(com.baytrendy.shopifyapp.utils.GraphQLResponse reponse) {
    }
    
    public final void setCurrencyData(@org.jetbrains.annotations.Nullable()
    java.lang.String currencyCode) {
    }
    
    public final void logOut() {
    }
    
    public final void insertPreviewData(@org.jetbrains.annotations.NotNull()
    org.json.JSONObject data) {
    }
    
    public LeftMenuViewModel(@org.jetbrains.annotations.NotNull()
    com.baytrendy.shopifyapp.repositories.Repository repository) {
        super();
    }
}

File: base/src/main/java/com/baytrendy/shopifyapp/basesection/viewmodels/LeftMenuViewModel.kt
package com.baytrendy.shopifyapp.basesection.viewmodels

import android.content.Context
import android.os.Handler
import android.os.Looper
import android.provider.Settings
import android.util.Log
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.shopify.buy3.GraphCallResult
import com.shopify.buy3.Storefront
import com.shopify.graphql.support.Error
import com.baytrendy.shopifyapp.MyApplication
import com.baytrendy.shopifyapp.R
import com.baytrendy.shopifyapp.dbconnection.entities.AppLocalData
import com.baytrendy.shopifyapp.dbconnection.entities.LivePreviewData
import com.baytrendy.shopifyapp.network_transaction.CustomResponse
import com.baytrendy.shopifyapp.network_transaction.doGraphQLQueryGraph
import com.baytrendy.shopifyapp.repositories.Repository
import com.baytrendy.shopifyapp.shopifyqueries.Query
import com.baytrendy.shopifyapp.utils.ApiResponse
import com.baytrendy.shopifyapp.utils.GraphQLResponse
import com.baytrendy.shopifyapp.utils.Status
import com.baytrendy.shopifyapp.utils.Urls

import java.util.HashMap
import java.util.Objects

import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.schedulers.Schedulers
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import org.json.JSONObject
import java.util.concurrent.Callable
import java.util.concurrent.Executors

class LeftMenuViewModel(var repository: Repository) : ViewModel() {
    private val disposables = CompositeDisposable()
    private val responseLiveData = MutableLiveData<ApiResponse>()
    val message = MutableLiveData<String>()
    val data = MutableLiveData<HashMap<String, String>>()
    private val currencyResponseLiveData = MutableLiveData<List<Storefront.CurrencyCode>>()
    private val handler = Handler()
    var context: Context? = null
    val isLoggedIn: Boolean
        get() = repository.isLogin

    fun Response(): MutableLiveData<ApiResponse> {
        getMenus()
        return responseLiveData
    }

    var customer_id: String
        get() = repository.customerId
        set(value) {}

    val cartCount: Int
        get() {
            val count = intArrayOf(0)
            try {
                val executor = Executors.newSingleThreadExecutor()
                val callable = Callable {
                    if (repository.allCartItems.size > 0) {
                        count[0] = repository.allCartItems.size
                    }
                    count[0]
                }
                val future = executor.submit(callable)
                count[0] = future.get()
                executor.shutdown()
            } catch (e: Exception) {
                e.printStackTrace()
            }

            return count[0]
        }
    val wishListcount: Int
        get() {
            val count = intArrayOf(0)
            try {
                val executor = Executors.newSingleThreadExecutor()
                val callable = Callable {
                    if (repository.wishListData.size > 0) {
                        count[0] = repository.wishListData.size
                    }
                    count[0]
                }
                val future = executor.submit(callable)
                count[0] = future.get()
                executor.shutdown()
            } catch (e: Exception) {
                e.printStackTrace()
            }

            return count[0]
        }

    fun fetchUserData() {
        try {
            val runnable = Runnable {
                val hashdata = HashMap<String, String>()
                if (repository.isLogin) {
                    val localData = repository.allUserData[0]
                    hashdata.put("firstname", localData.firstname!!)
                    hashdata.put("secondname", localData.lastname!!)
                    hashdata.put("tag", "login")
                    Log.i("MageNative", "LeftMenuResume 2" + localData.firstname!!)

                } else {
                    Log.i("MageNative", "LeftMenuResume 2" + "Sign")
                    hashdata["firstname"] = context?.getString(R.string.sign_first)!!
                    hashdata["secondname"] = context?.getString(R.string.in_last)!!
                    hashdata["tag"] = "Sign In"
                }
                handler.post { data.value = hashdata }
            }
            Thread(runnable).start()
        } catch (e: Exception) {
            e.printStackTrace()
        }


    }

    private fun getMenus() {
        try {
            disposables.add(repository.getMenus(Urls(MyApplication.context)!!.mid)
                    .subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())
                    .subscribe(
                            { result -> responseLiveData.setValue(ApiResponse.success(result)) },
                            { throwable -> responseLiveData.setValue(ApiResponse.error(throwable)) }
                    ))
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    fun currencyResponse(): MutableLiveData<List<Storefront.CurrencyCode>> {
        getCurrency()
        return currencyResponseLiveData
    }

    private fun getCurrency() {
        try {
            doGraphQLQueryGraph(repository, Query.shopDetails, customResponse = object : CustomResponse {
                override fun onSuccessQuery(result: GraphCallResult<Storefront.QueryRoot>) {
                    invoke(result)
                }
            }, context = context!!)

        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    override fun onCleared() {
        disposables.clear()
    }

    private operator fun invoke(result: GraphCallResult<Storefront.QueryRoot>): Unit {
        if (result is GraphCallResult.Success<*>) {
            consumeResponse(GraphQLResponse.success(result as GraphCallResult.Success<*>))
        } else {
            consumeResponse(GraphQLResponse.error(result as GraphCallResult.Failure))
        }
        return Unit
    }

    private fun consumeResponse(reponse: GraphQLResponse) {
        when (reponse.status) {
            Status.SUCCESS -> {
                val result = (reponse.data as GraphCallResult.Success<Storefront.QueryRoot>).response
                if (result.hasErrors) {
                    val errors = result.errors
                    val iterator = errors.iterator()
                    val errormessage = StringBuilder()
                    var error: Error? = null
                    while (iterator.hasNext()) {
                        error = iterator.next()
                        errormessage.append(error.message())
                    }
                    message.setValue(errormessage.toString())
                } else {
                    currencyResponseLiveData.setValue(Objects.requireNonNull<Storefront.QueryRoot>(result.data).getShop().getPaymentSettings().getEnabledPresentmentCurrencies())
                }
            }
            Status.ERROR -> message.setValue(reponse.error!!.error.message)
            else -> {
            }
        }
    }

    fun setCurrencyData(currencyCode: String?) {
        val runnable = Runnable {
            val appLocalData = repository.localData[0]
            appLocalData.currencycode = currencyCode
            repository.updateData(appLocalData)
        }
        Thread(runnable).start()

    }

    fun logOut() {
        val runnable = Runnable {
            Log.i("MageNative", "LeftMenuResume 5")
            repository.deletecart()
            repository.deleteWishListData()
            repository.deleteUserData()
            fetchUserData()
        }
        Thread(runnable).start()
    }

    fun deletLocal() {
        GlobalScope.launch(Dispatchers.IO) {
            repository.deleteLocalData()
        }
    }

    fun insertPreviewData(data: JSONObject) {
        val runnable = Runnable {
            var lpreview = repository.getPreviewData()
            if (lpreview.size == 0) {
                var preview = LivePreviewData(data.getString("mid"), data.getString("shopUrl"), data.getString("token"))
                repository.insertPreviewData(preview)
            } else {
                var preview = lpreview.get(0)
                preview.mid = data.getString("mid")
                preview.shopurl = data.getString("shopUrl")
                preview.apikey = data.getString("token")
                repository.updatePreviewData(preview)
            }
        }
        Thread(runnable).start()
    }
}
